config/ImathConfig.h.in

```diff
@@ -174,4 +174,9 @@
 #    endif
 #endif
 
+// Whether the copy and move constructors of Vec2, Vec3 and Vec4 are default or not.
+#ifndef IMATH_VEC_BE_TRIVIALLY_RELOCATABLE
+#     define IMATH_VEC_BE_TRIVIALLY_RELOCATABLE 0
+#endif
```

Comment:
Author of P1144 here. :) I think it's great you're adopting the terminology — and orthogonally great that you're `=default`'ing all the things — but, looking at this PR, I think the focus on the word "relocatable" is actually a distraction. You should just say `IMATH_VEC_RULE_OF_ZERO` or something like that. After your changes, it is true that a `Vec2<T>` will be trivially relocatable if `T` is trivially relocatable; but it will also be trivially copyable if `T` is trivially copyable; trivially copy-assignable if `T` is trivially copy-assignable; and so on. There's nothing relocation-specific about this PR as far as I can tell. (Am I missing something?)

Followup: Why is this gated on a macro at all? The `=default` changes look like a strict improvement to me. If the non-defaulted implementations were in a .cpp file, so they were only compiled once, then there might be a reason to keep them; but nope, they're `inline constexpr` in the header! I think you should consider very hard whether there's any reason to ever set the macro to anything but `1`. (...Is it that you need to have an opt-in way to avoid the ABI break that comes with Vec2 suddenly being passed in registers?)

Reply:
FYI @sukeya @cary-ilm, there is _right now this week_ a PR to add fully-P1144-compliant `__is_trivially_relocatable(T)` to Clang (fixing a long-standing false-positive bug on Windows and also making it respect user-defined special members including `operator=` so as to avoid false positives on all platforms). If you're interested (professionally, financially, or just plain interested) in P1144 trivial relocatability, I'd really appreciate it if you'd take a look at https://github.com/llvm/llvm-project/pull/84621 and weigh in with your thoughts. (Full disclosure: my impression is that the false-positives being fixed don't actually matter to @sukeya's use-case of `thrust::async::copy`. But I'd still really appreciate your taking a look, if you have interest in moving P1144 forward at all. It [and I mean that PR, specifically!] needs to see some customer demand in order to progress.)

Reply:
Thank you for your advice! I committed your patches.
I'm busy now, but I'll see that PR when I have a time.

Reply:
@sukeya Thank you!  I'd really appreciate it if you leave a :+1: reaction _now_, so you'll receive notifications that remind you to add a fuller comment when you have the time.